### ISSUE TITLE
The loss doesn't decrease when I want to fit a texture map in TexturesUV

### ISSUE THREAD

**Issue by JasonBoy1** [facebookresearch/pytorch3d#1008]
## 🐛 Bugs / Unexpected behaviors

Here is the code, I simply changed the code of your [tutorials](https://github.com/facebookresearch/pytorch3d/blob/main/docs/tutorials/fit_textured_mesh.ipynb).
I changed the textures of the new_src_mesh to TexturesUV and initialize the texture_map with zeros.
The loss never decreases, and the texture map remains all black after the fitting. 
Could anybody help me?

```python
import os
import torch
import matplotlib.pyplot as plt

from pytorch3d.utils import ico_sphere
import numpy as np
from tqdm.notebook import tqdm

# Util function for loading meshes
from pytorch3d.io import load_objs_as_meshes, save_obj

from pytorch3d.loss import (
    chamfer_distance, 
    mesh_edge_loss, 
    mesh_laplacian_smoothing, 
    mesh_normal_consistency,
)

# Data structures and functions for rendering
from pytorch3d.structures import Meshes
from pytorch3d.renderer import (
    look_at_view_transform,
    OpenGLPerspectiveCameras, 
    PointLights, 
    DirectionalLights, 
    Materials, 
    RasterizationSettings, 
    MeshRenderer, 
    MeshRasterizer,  
    SoftPhongShader,
    SoftSilhouetteShader,
    SoftPhongShader,
    TexturesVertex,
    TexturesUV
)

# add path for demo utils functions 
import sys
import os
sys.path.append(os.path.abspath(''))

# Rasterization settings for differentiable rendering, where the blur_radius
# initialization is based on Liu et al, 'Soft Rasterizer: A Differentiable 
# Renderer for Image-based 3D Reasoning', ICCV 2019
sigma = 1e-4
raster_settings_soft = RasterizationSettings(
    image_size=128, 
    blur_radius=0, 
    faces_per_pixel=1, 
    perspective_correct=False,
)
lights = PointLights(ambient_color=[[1.0, 1.0, 1.0]],
    diffuse_color=[[0, 0, 0]],
    specular_color=[[0, 0, 0]],
    location=[[0, 0, 1.0]],).to(device)
blend_params = BlendParams(sigma=1e-4, gamma=1e-4, background_color=(1, 1, 1.0))
# Differentiable soft renderer using per vertex RGB colors for texture
renderer_textured = MeshRenderer(
    rasterizer=MeshRasterizer(
        cameras=camera, 
        raster_settings=raster_settings_soft
    ),
    shader=SoftPhongShader(device=device, 
        cameras=camera,
        lights=lights,
       blend_params=blend_params
                          )
)

# Number of views to optimize over in each SGD iteration
num_views_per_iteration = 2
# Number of optimization steps
Niter = 2000
# Plot period for the losses
plot_period = 250

%matplotlib inline
from pytorch3d.renderer import TexturesUV
# Optimize using rendered RGB image loss, rendered silhouette image loss, mesh 
# edge loss, mesh normal consistency, and mesh laplacian smoothing
losses = {"rgb": {"weight": 1.0, "values": []},
          "silhouette": {"weight": 0., "values": []},
          "edge": {"weight": 0., "values": []},
          "normal": {"weight": 0., "values": []},
          "laplacian": {"weight": .0, "values": []},
         }

# We will learn to deform the source mesh by offsetting its vertices
# The shape of the deform parameters is equal to the total number of vertices in 
# src_mesh
src_mesh = load_objs_as_meshes(['./data/cow_mesh/cow.obj']).to(device)
verts_shape = src_mesh.verts_packed().shape
# deform_verts = torch.full(verts_shape, 0.0, device=device, requires_grad=True)

# We will also learn per vertex colors for our sphere mesh that define texture 
# of the mesh
# sphere_verts_rgb = torch.full([1, verts_shape[0], 3], 0.5, device=device, requires_grad=True)

shpere_texture_image = torch.full([1, 512, 512, 3], 0., device=device, requires_grad=True)
# The optimizer
optimizer = torch.optim.SGD([shpere_texture_image], lr=2.0, momentum=0.9)

loop = tqdm(range(Niter))

for i in loop:
    # Initialize optimizer
    optimizer.zero_grad()
    
    # Deform the mesh
#     new_src_mesh = src_mesh.offset_verts(deform_verts)
    new_src_mesh = src_mesh.clone()
    # Add per vertex colors to texture the mesh
#     new_src_mesh.textures = TexturesVertex(verts_features=sphere_verts_rgb) 
    new_src_mesh.textures._maps_padded = shpere_texture_image
    # Losses to smooth /regularize the mesh shape
    loss = {k: torch.tensor(0.0, device=device) for k in losses}
    update_mesh_shape_prior_losses(new_src_mesh, loss)
    
    # Randomly select two views to optimize over in this iteration.  Compared
    # to using just one view, this helps resolve ambiguities between updating
    # mesh shape vs. updating mesh texture
    for j in np.random.permutation(num_views).tolist()[:num_views_per_iteration]:
        images_predicted = renderer_textured(new_src_mesh, cameras=target_cameras[j], lights=lights)

        # Squared L2 distance between the predicted silhouette and the target 
        # silhouette from our dataset
        predicted_silhouette = images_predicted[..., 3]
        loss_silhouette = ((predicted_silhouette - target_silhouette[j]) ** 2).mean()
#         loss["silhouette"] += loss_silhouette / num_views_per_iteration
        
        # Squared L2 distance between the predicted RGB image and the target 
        # image from our dataset
        predicted_rgb = images_predicted[..., :3]
        loss_rgb = ((predicted_rgb - target_rgb[j]) ** 2).mean()
        loss["rgb"] += loss_rgb / num_views_per_iteration
    
    # Weighted sum of the losses
    sum_loss = torch.tensor(0.0, device=device)
    for k, l in loss.items():
        sum_loss += l * losses[k]["weight"]
        losses[k]["values"].append(float(l.detach().cpu()))
    
    # Print the losses
    loop.set_description("total_loss = %.6f" % sum_loss)
    
    # Plot mesh
    if i % plot_period == 0:
        visualize_prediction(new_src_mesh, renderer=renderer_textured, title="iter: %d" % i, silhouette=False)
        
    # Optimization step
    sum_loss.backward()
    optimizer.step()

```

**Comment by JasonBoy1** [comment]
Ok, I raised up the lr and get some colors, but the rendered images look very weird.
Seems that the eye of cow has some Ghosting.

![image](https://user-images.githubusercontent.com/33180378/147488885-2fea9718-d3b0-4432-b817-70984a2a6c27.png)

**Comment by Dawars** [comment]
Did you manage to fix the ghosting?

**Comment by xk-huang** [comment]
I've encountered the same problem. After changing the optimizer from SGD to Adam, the texture of the mesh is fitted correctly.

**Comment by blacksino** [comment]
> I've encountered the same problem. After changing the optimizer from SGD to Adam, the texture of the mesh is fitted correctly.

quite curious about the reason
